pkgs/tools/networking/edgedb/default.nix

```diff
@@ -0,0 +1,55 @@
+{ stdenv
+, lib
+, runCommand
+, patchelf
+, fetchFromGitHub
+, rustPlatform
+, makeWrapper
+, pkg-config
+, curl
+, Security
+, CoreServices
+, libiconv
+, xz
+, perl
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "edgedb";
+  version = "1.2.0-unstable";
+
+  src = fetchFromGitHub {
+    owner = "edgedb";
+    repo = "edgedb-cli";
+    rev = "3c65c8bf0a09988356ad477d0ae234182f809b0a";
+    sha256 = "sha256-UqoRa5ZbPJEHo9wyyygrN1ssorgY3cLw/mMrCDXr4gw=";
+  };
+
+  cargoSha256 = "sha256-Axzlk6k1du6gws8sYaFcLoKOq+rL4ngmEBga5vvdKws=";
+
+  nativeBuildInputs = [ makeWrapper pkg-config perl ];
+
+  buildInputs = [
+    curl
+  ] ++ lib.optionals stdenv.isDarwin [ CoreServices Security libiconv xz ];
+
+  checkFeatures = [ ];
+
+  patches = lib.optionals stdenv.isLinux [
+    (runCommand "0001-dynamically-patchelf-binaries.patch" { CC = stdenv.cc; patchelf = patchelf; } ''
+      export dynamicLinker=$(cat $CC/nix-support/dynamic-linker)
+      substitute ${./0001-dynamically-patchelf-binaries.patch} $out \
+        --subst-var patchelf \
+        --subst-var dynamicLinker \
+    '')
```

Comment:
We should be able todo this without runCommand and just substitute. The path to the dynamic linker is hardcoded and can be derived from stdenv.cc.